tutorials/ExploringDeutschJozsaAlgorithm/DeutschJozsaAlgorithmTutorial_P1.ipynb

```diff
@@ -0,0 +1,189 @@
+    "\n",
+    "This tutorial will:\n",
+    "* introduce you to the problems solved by the Deutsch and Deutsch–Jozsa algorithms and walk you through the classical solution to them,\n",
+    "* give you a brief introduction to quantum oracles (for a more detailed introduction, see the [Oracles tutorial](../Oracles/Oracles.ipynb)),\n",
+    "* describe the idea behind the Deutsch and Deutsch–Jozsa algorithms and walk you through the math for them,\n",
+    "* teach you how to implement the algorithms in the Q# programming language,\n",
+    "* and finally help you to run your implementation of the algorithm on several quantum oracles to see for yourself how the algorithm works!\n",
+    "\n",
+    "Let's go!\n",
+    "\n",
+    "This tutorial consists of several notebooks:\n",
+    "1. Part I covers the problem statement and the classical algorithm for solving it.\n",
+    "2. [Part II](./DeutschJozsaAlgorithmTutorial_P2.ipynb) introduces single-qubit quantum oracles and Deutsch algorithm for solving the problem for one-bit input functions.\n",
+    "3. [Part III](./DeutschJozsaAlgorithmTutorial_P3.ipynb) introduces multi-qubit quantum oracles and Deutsch-Jozsa algorithm for solving the general case of the problem."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part I. Problem statement and classical algorithm\n",
+    "\n",
+    "## The problem\n",
+    "\n",
+    "You are given a classical function $f(x): \\{0, 1\\}^N \\to \\{0, 1\\}$. You are guaranteed that the function $f$ is\n",
+    "* either *constant* (has the same value for all inputs) \n",
+    "* or *balanced* (has value 0 for half of the inputs and 1 for the other half of the inputs). \n",
+    "\n",
+    "The task is to figure out whether the function is constant or balanced.\n",
+    "\n",
+    "For the single-bit functions ($N = 1$) the problem can be formulated in a simpler way: given a classical function $f(x): \\{0, 1\\} \\to \\{0, 1\\}$, figure out whether $f(0) = f(1)$. \n",
+    "Indeed, $f(0) = f(1)$ is an equivalent definition of a constant single-bit function, and $f(0) \\neq f(1)$ - of a balanced single-bit function.\n",
+    "\n",
+    "> ## Examples\n",
+    ">\n",
+    "> * $f(x) \\equiv 0$ or $f(x) \\equiv 1$ are examples of constant functions (and they are actually the only constant functions in existence).\n",
+    "> * For single-bit functions, $f(x) = x$ and $f(x) = 1 - x$ are the only existing balanced functions.\n",
+    "> * $f(x) = x \\text{ mod } 2$ (the least significant bit of $x$) or $f(x) = 1 \\text{ if the binary notation of }x \\text{ has odd number of 1s and 0 otherwise}$ are examples of multi-bit balanced functions.  \n",
```

Comment:
Does `\bmod` work instead of `\text{ mod }`?

Reply:
It does! I replaced it in all notebooks